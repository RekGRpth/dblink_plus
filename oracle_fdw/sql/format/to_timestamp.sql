\c contrib_regression_utf8

SET datestyle TO ISO;
SET timezone TO 'Asia/Tokyo';

SELECT to_timestamp(999999999999223372036854.77442);
SELECT to_timestamp(9223372036854.77442);
SELECT to_timestamp(9223372036854.77441999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999);
SELECT to_timestamp(9223372036854.7744);

SELECT id, to_date(val1, val2) FROM ft_varchar2_b_varchar ORDER BY id;

--  to_timestamp(text,text) char_b
EXPLAIN (COSTS FALSE) 
WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
EXPLAIN (COSTS FALSE) 
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-01'::timestamptz ORDER BY id;
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-01'::timestamptz ORDER BY id;

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-11-01 BC'::timestamptz ORDER BY id;
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-11-01 BC'::timestamptz ORDER BY id;

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-01-11 BC'::timestamptz ORDER BY id;
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-01-11 BC'::timestamptz ORDER BY id;

