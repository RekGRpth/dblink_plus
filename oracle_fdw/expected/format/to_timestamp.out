\c contrib_regression_utf8
SET datestyle TO ISO;
SET timezone TO 'Asia/Tokyo';
SELECT to_timestamp(999999999999223372036854.77442);
          to_timestamp           
---------------------------------
 294247-01-10 13:00:54.775808+09
(1 row)

SELECT to_timestamp(9223372036854.77442);
          to_timestamp           
---------------------------------
 294247-01-10 13:00:54.775808+09
(1 row)

SELECT to_timestamp(9223372036854.77441999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999);
          to_timestamp          
--------------------------------
 294247-01-10 13:00:54.77376+09
(1 row)

SELECT to_timestamp(9223372036854.7744);
          to_timestamp          
--------------------------------
 294247-01-10 13:00:54.77376+09
(1 row)

SELECT id, to_date(val1, val2) FROM ft_varchar2_b_varchar ORDER BY id;
 id |    to_date    
----+---------------
  1 | 
  2 | 
  3 | 2011-01-02
  4 | 2011-01-02
  5 | 2011-01-02
  6 | 2011-01-02
  7 | 2011-01-02
  8 | 2011-01-02
  9 | 2011-01-01
 10 | 0001-11-01 BC
 11 | 0001-01-11 BC
(11 rows)

--  to_timestamp(text,text) char_b
EXPLAIN (COSTS FALSE) 
WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
                                                   QUERY PLAN                                                    
-----------------------------------------------------------------------------------------------------------------
 Sort
   Sort Key: ft_varchar2_b_varchar.id
   CTE ft_varchar2_b_varchar
     ->  Foreign Scan on ft_varchar2_b_varchar
           Remote SQL: SELECT "id", "val1", "val2" FROM "oracle_fdw"."t_varchar2_b"
   ->  CTE Scan on ft_varchar2_b_varchar
         Filter: (to_timestamp((val1)::text, (val2)::text) = '2011-01-02 00:00:00+09'::timestamp with time zone)
(7 rows)

EXPLAIN (COSTS FALSE) 
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
                                                                                                       QUERY PLAN                                                                                                       
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sort
   Sort Key: id
   ->  Foreign Scan on ft_varchar2_b_varchar
         Remote SQL: SELECT "id", "val1", "val2" FROM "oracle_fdw"."t_varchar2_b" WHERE (TO_TIMESTAMP("val1","val2") = TO_TIMESTAMP_TZ('2011-01-02 AD 00:00:00.000000 Asia/Tokyo', 'YYYY-MM-DD BC HH24:MI:SS.FF6 TZR'))
(4 rows)

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
 id |     val1     |     val2     
----+--------------+--------------
  3 | 2011-01-02   | YYYY-MM-DD
  4 | 2011:02:01   | YYYY:DD:MM
  5 | 01--2011--02 | MM--YYYY--DD
  6 | 01::02::2011 | MM::DD::YYYY
  7 | 02--2011--01 | DD--YYYY--MM
  8 | 02::01::2011 | DD::MM::YYYY
(6 rows)

SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-02'::timestamptz ORDER BY id;
 id |     val1     |     val2     
----+--------------+--------------
  3 | 2011-01-02   | YYYY-MM-DD
  4 | 2011:02:01   | YYYY:DD:MM
  5 | 01--2011--02 | MM--YYYY--DD
  6 | 01::02::2011 | MM::DD::YYYY
  7 | 02--2011--01 | DD--YYYY--MM
  8 | 02::01::2011 | DD::MM::YYYY
(6 rows)

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-01'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
  9 | 2011 | YYYY
(1 row)

SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '2011-01-01'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
(0 rows)

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-11-01 BC'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
 10 | 11   | MM
(1 row)

SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-11-01 BC'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
(0 rows)

WITH ft_varchar2_b_varchar AS (SELECT * FROM ft_varchar2_b_varchar)
SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-01-11 BC'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
 11 | 11   | DD
(1 row)

SELECT * FROM ft_varchar2_b_varchar WHERE to_timestamp(val1, val2) = '0001-01-11 BC'::timestamptz ORDER BY id;
 id | val1 | val2 
----+------+------
(0 rows)

